### PR TITLE
Add AWS feature to launch Elastic Transcoder

### PR DESCRIPTION
**Pull request by aiotph**
Hello Gauthier :)

Take a look please ! God of code mwahahah ^^

### CODE REVIEW

File: infra/deploy/main.tf [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+assume_role_policy  = <<EOF
+{
+    "Version": "2012-10-17",
+    "Statement": [
+    {
+        "Action": "sts:AssumeRole",
+        "Principal": {
+            "Service": "elastictranscoder.amazonaws.com"
+        },
+    "Effect": "Allow"
```
Comment (Ninir) [inline infra/deploy/main.tf]: Indentation 😄

File: infra/deploy/main.tf [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+
+#
+# AWS CREATE INPUT & OUTPUT FOLDERS
+#
+resource "aws_s3_bucket_object" "input_folder" {
+    bucket = "${aws_s3_bucket.main.id}"
+    acl    = "private"
+    key    = "Input/"
+    source = "/dev/null"
+}
```
Comment (Ninir) [inline infra/deploy/main.tf]: These bucket objects are not needed: as S3 object keys are simply keys (thus not being directories), creating a file to `input/plop/toto/foo/bar/baz` is totally valid

**Review by Ninir — COMMENTED** [review]
Hey @tphed2si 

Looks good at first sight :)

Could you also run `terraform fmt`? it will format the terraform code so it's all normalized! :)

**Comment by aiotph** [comment]
Already done. Coming to the next push ! 😉


Tony PHE
-------- Original message --------From: Gauthier Wallet <notifications@github.com> Date: 06/11/2017  12:02  (GMT+01:00) To: d2si/AutoSub <AutoSub@noreply.github.com> Cc: tphed2si <tony.phe@d2-si.eu>, Mention <mention@noreply.github.com> Subject: Re: [d2si/AutoSub] Add AWS feature to launch Elastic Transcoder (#1) 
@Ninir commented on this pull request.

Hey @tphed2si
Looks good at first sight :)
Could you also run terraform fmt? it will format the terraform code so it's all normalized! :)

In infra/deploy/main.tf:
> +# AWS ROLE
+#
+# TRANSCODER ROLE
+resource "aws_iam_role" "iam_role_transcoder" {
+    name = "transcoder_role"
+
+assume_role_policy  = <<EOF
+{
+    "Version": "2012-10-17",
+    "Statement": [
+    {
+        "Action": "sts:AssumeRole",
+        "Principal": {
+            "Service": "elastictranscoder.amazonaws.com"
+        },
+    "Effect": "Allow"

Indentation 😄



In infra/deploy/main.tf:
> +  lambda_function {
+    lambda_function_arn = "${aws_lambda_function.lambda.arn}"
+    events              = ["s3:ObjectCreated:*"]
+    filter_prefix       = "Input/"
+  }
+}
+
+#
+# AWS CREATE INPUT & OUTPUT FOLDERS
+#
+resource "aws_s3_bucket_object" "input_folder" {
+    bucket = "${aws_s3_bucket.main.id}"
+    acl    = "private"
+    key    = "Input/"
+    source = "/dev/null"
+}

These bucket objects are not needed: as S3 object keys are simply keys (thus not being directories), creating a file to input/plop/toto/foo/bar/baz is totally valid

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or mute the thread.


  
  




{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/d2si/AutoSub","title":"d2si/AutoSub","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/d2si/AutoSub"}},"updates":{"snippets":[{"icon":"PERSON","message":"@Ninir commented on #1"}],"action":{"name":"View Pull Request","url":"https://github.com/d2si/AutoSub/pull/1#pullrequestreview-74382112"}}}

File: infra/deploy/main.tf (hunk 2 above)
Comment (aiotph) [reply infra/deploy/main.tf]: I thought it was better to get an object key for the directories I guess just to make sure that people will understand it well.

File: infra/deploy/main.tf (hunk 1 above)
Comment (aiotph) [reply infra/deploy/main.tf]: Done ! Applied on my last push.

**Comment by Ninir** [comment]
Closing in favor of #2 , based on this work

**Pull request closed without merging**